libs/server/Resp/Parser/RespCommand.cs

```diff
@@ -321,17 +321,52 @@ cmd switch
                 };
         }
 
+        public static RespCommand FirstReadCommand()
+            => RespCommand.NONE + 1;
+
+        public static RespCommand LastReadCommand()
+            => RespCommand.APPEND - 1;
+
+        public static RespCommand FirstWriteCommand()
+            => RespCommand.APPEND;
+
+        public static RespCommand LastWriteCommand()
+            => RespCommand.BITOP_NOT;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsReadOnly(this RespCommand cmd)
+            => cmd <= LastReadCommand();
+
+        public static bool IsDataCommand(this RespCommand cmd)
+        {
+            return cmd switch
+            {
+                // TODO: validate if these cases need to be excluded
+                RespCommand.MIGRATE => false,
+                RespCommand.DBSIZE => false,
+                RespCommand.MEMORY_USAGE => false,
+                RespCommand.FLUSHDB => false,
+                _ => cmd >= FirstReadCommand() && cmd <= LastWriteCommand()
+            };
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsWriteOnly(this RespCommand cmd)
```

Comment:
This approach seems fairly error-prone to me. Have we thought about alternative approaches? Maybe adding a read-only flag?

Reply:
This piece was there from the ACL refactor; I just moved it under a common method for people to re-use it. I am actually using the IsReadOnly method myself. Can we have IsReadOnly as part of the command info instead?